model-optimizer/extensions/middle/MarkSubgraphsWithCorrectLayout.py

```diff
@@ -121,3 +125,88 @@ def find_and_replace_pattern(self, graph: Graph):
             for visited_node in marked_nodes:
                 mark_as_correct_data_layout(visited_node)
                 visited_node['nchw_layout'] = True
+
+        for node in self.get_ports_and_nodes_on_weights(graph)[1]:
+            mark_as_correct_data_layout(node)
+            node['nchw_layout'] = True
+            if node.soft_get('type') == 'Const':  # WA for Const op deletion during clean_up
+                node.out_node()['nchw_layout'] = True
+
+        for node in self.get_ports_and_nodes_on_shape_subgraphs(graph)[1]:
+            mark_as_correct_data_layout(node)
+            node['nchw_layout'] = True
+            if node.soft_get('type') == 'Const':  # WA for Const op deletion during clean_up
+                node.out_node()['nchw_layout'] = True
+
+    @staticmethod
+    def walk_up_from_in_ports_to_out_ports(in_ports: Set[Port], out_ports: Set[Port],
```

Comment:
please, verify that this graph traversing does not work forever for a recent model with Split/Concat with 800 ports.

Reply:
I checked and it doesn't.